File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/XaPhuongColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { XaPhuongRow } from "./XaPhuongRow";

export interface XaPhuongColumns {
    Id: Column<XaPhuongRow>;
    TenXaPhuong: Column<XaPhuongRow>;
    QuanHuyenTen: Column<XaPhuongRow>;
}

export class XaPhuongColumns extends ColumnsBase<XaPhuongRow> {
    static readonly columnsKey = 'SanPhamDB.XaPhuong';
    static readonly Fields = fieldsProxy<XaPhuongColumns>();
}

File: wheystore/wheystore.Web/Modules/SanPhamDB/XaPhuong/XaPhuongColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.XaPhuong")]
[BasedOnRow(typeof(XaPhuongRow), CheckNames = true)]
public class XaPhuongColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string TenXaPhuong { get; set; }

    [DisplayName("Quận Huyện")]
    public string QuanHuyenTen { get; set; }
}